### ISSUE TITLE
Organizing/modeling Views

### ISSUE THREAD

**Issue by felixbecker** [lovoo/goka#32]
Hi, 

I have a question regarding the example-2 which pobpulates an rest api so one can access the number of clicks for a given user-id as key.
How would I implement a list of users with their number of clicks? ( basically a View.all() if this would even make sense ). 
Or Would this be something I have to manage in a processor function? Like a slice of users with their id and clicks? Or is there a way to iterate over the keys? What would be the recommended way?

Any plans to extend the docs or examples for more advanced scenarios covering Lookup, Join, Loopback, etc?

Thanks for your feedback in advance.

**Comment by db7** [comment]
To iterate over all key-value pairs of a table, we use `View.Iterator()` which should be pretty soon merged to the master.

And yes, we do have plans to extend the docs. Have you checked this [blog post](https://tech.lovoo.com/2017/05/23/goka/)? We should probably link it in the README at some point.

**Comment by felixbecker** [comment]
Hi, thanks for your reply. I think `View.Iterator()` interface would be a great fit. 
I saw the blog post and I am looking forward to other posts in that series since it made me curious for more.

**Comment by SamiHiltunen** [comment]
Hey @felixbecker, we just merged #33. This gives you access to an iterator that iterates over the whole view. Let me know if any issues arise. Example assuming string values:

```go
state, err = view.Iterator()
if err != nil {
  log.Fatalf("error getting iterator: %v", err)
}
defer state.Release()

for state.Next() {
  key := state.Key()
  val, err := state.Value()
  if err != nil {
    log.Fatalf("error getting value from iterator: %v", err)
  }

  log.Printf("key: %s, value: %s", string(key), val.(string))
}
```
